cli/provider/cmd.go

```diff
@@ -616,7 +617,11 @@ func (c *CmdConfigurator) ValidateFlags(ctx context.Context, cmd *cobra.Command)
 func (c *CmdConfigurator) CreateConfigFile(ctx context.Context, defaultCfg config.Config) error {
 	defaultCfg = c.UpdateConfigData(defaultCfg)
 	toolSvc := tools.NewTools(c.logger, nil)
-	configData := defaultCfg
+	configData, err := c.RemoveInternalParameters(defaultCfg)
```

Comment:
you can combine RemoveInternalParameters and yaml marshal into one function.

Reply:
Done